### PR TITLE
feat: Extend `Frame::Padding` with length

### PR DESCRIPTION
**Pull request by larseggert**
This is preparation of qlog supporting the logging of runs of padding frames via `payload_length`, instead of each one individually.

Needs https://github.com/cloudflare/quiche/pull/1745 (eventually).

### CODE REVIEW

File: neqo-transport/src/packet/mod.rs [hunk 1]
```diff
@@ -257,7 +258,8 @@ impl PacketBuilder {
     /// Returns true if padding was added.
     pub fn pad(&mut self) -> bool {
         if self.padding && !self.is_long() {
-            self.encoder.pad_to(self.limit, 0);
+            self.encoder
+                .pad_to(self.limit, FRAME_TYPE_PADDING.try_into().unwrap());
```
Comment (larseggert) [inline neqo-transport/src/packet/mod.rs:262]: I can back this out, but I found it weird that we hard-coded the frame type here.

File: neqo-transport/src/frame.rs [hunk 2]
```diff
@@ -103,7 +103,9 @@ pub struct AckRange {
 
 #[derive(PartialEq, Eq, Debug, Clone)]
 pub enum Frame<'a> {
-    Padding,
+    Padding {
+        length: u64,
+    },
```
Comment (martinthomson) [inline neqo-transport/src/frame.rs:108]: Padding(u64) would be OK.  But I question the need for a u64 here.  Wouldn't u16 work?

File: neqo-transport/src/frame.rs [hunk 3]
```diff
@@ -409,13 +409,24 @@ impl<'a> Frame<'a> {
...
-        let t = d(dec.decode_varint())?;
+        let t = dv(dec)?;
         match t {
-            FRAME_TYPE_PADDING => Ok(Self::Padding),
+            FRAME_TYPE_PADDING => {
+                let mut length = 1;
+                // TODO: It may be worthwhile to find a more efficient way to skip
+                // over runs of zero bytes.
+                while dec.remaining() > 0
+                    && u64::from(dec.peek_byte().unwrap()) == FRAME_TYPE_PADDING
```
Comment (martinthomson) [inline neqo-transport/src/frame.rs:419]: Can we maybe have a peek() that returns `Option<u8>`?

File: neqo-transport/src/packet/mod.rs (hunk 1 above)
Comment (martinthomson) [reply neqo-transport/src/packet/mod.rs:262]: Can you use `Frame::Padding(1).get_type()`?  That should optimize well enough.

File: neqo-transport/src/qlog.rs [hunk 4]
```diff
@@ -395,7 +395,7 @@ pub fn metrics_updated(qlog: &mut NeqoQlog, updated_metrics: &[QlogMetric]) {
 #[allow(clippy::cast_possible_truncation, clippy::cast_precision_loss)] // No choice here.
 fn frame_to_qlogframe(frame: &Frame) -> QuicFrame {
```
Comment (martinthomson) [inline neqo-transport/src/qlog.rs:396]: Hmm, why are we not implementing `From<&Frame>` for `QuicFrame`?

**Review by martinthomson — APPROVED** [review]
Looks good.

File: neqo-transport/src/frame.rs (hunk 3 above)
Comment (martinthomson) [reply neqo-transport/src/frame.rs:419]: Ok, we have one already.

Do this instead:

```rust
while let Some(b) == dec.peek_byte() {
    if b != FRAME_TYPE_PADDING { break; }
    len += 1;
    dec.skip(1);
}
```

File: neqo-transport/src/packet/mod.rs (hunk 1 above)
Comment (larseggert) [reply neqo-transport/src/packet/mod.rs:262]: Also returns `u64`, so would still need to convert.

Comment (martinthomson) [reply neqo-transport/src/packet/mod.rs:262]: Ah, of course.  Which is of course why I just put a raw 0 there before.  What you have is totally fine.

**Review by KershawChang — APPROVED** [review]
lgtm

**Pull request merged**